common/lib/dependabot/package_manager.rb

```diff
@@ -58,18 +58,36 @@ def deprecated?
       deprecated_versions.include?(version)
     end
 
+    # Raises an error if the current package manager version is unsupported.
+    # If the version is unsupported, it raises a ToolVersionNotSupported error.
+    sig { void }
+    def raise_if_unsupported!
+      return unless unsupported?
+
+      # Example: v2.*, v3.*
+      supported_versions_message = supported_versions.map { |v| "v#{v}.*" }.join(", ")
+
+      raise ToolVersionNotSupported.new(
+        name,
+        version.to_s,
+        supported_versions_message
+      )
+    end
+
     # Checks if the current version is unsupported.
     # Returns true if the version is in the unsupported_versions array; false otherwise.
     # @example
     #   package_manager.unsupported? #=> false
     sig { returns(T::Boolean) }
     def unsupported?
+      # If version specified is in the unsupported_versions array, return true.
       return true if unsupported_versions.include?(version)
 
       supported_versions = self.supported_versions
-      return version < supported_versions.first if supported_versions.any?
```

Comment:
This method is written when doing deprecation warning however it is not used since we didn't implemented yet at that time unsupported error. We are refining this method to fit the case